homeassistant/components/dyson.py

```diff
@@ -69,14 +69,17 @@ def setup(hass, config):
             dyson_device = next((d for d in dyson_devices if
                                  d.serial == device["device_id"]), None)
             if dyson_device:
-                connected = dyson_device.connect(None, device["device_ip"],
-                                                 timeout, retry)
-                if connected:
-                    _LOGGER.info("Connected to device %s", dyson_device)
-                    hass.data[DYSON_DEVICES].append(dyson_device)
-                else:
-                    _LOGGER.warning("Unable to connect to device %s",
-                                    dyson_device)
+                try:
+                    connected = dyson_device.connect(device["device_ip"])
```

Comment:
The first block is for manual connection (specifying device id/ip). The second one try to use mDNS protocol to discover fan/purifier devices (not working with robot vacuum yet). https://home-assistant.io/components/dyson/
It was the same method before but it was quite painful to use so I updated the underlying library and create 2 different methods.